pkg/network/usm/ebpf_gotls.go

```diff
@@ -124,58 +125,82 @@ type GoTLSProgram struct {
+				EBPFFuncName: connReadRetProbe,
+			},
+		},
+		{
+			ProbeIdentificationPair: manager.ProbeIdentificationPair{
+				EBPFFuncName: connWriteProbe,
+			},
+		},
+		{
+			ProbeIdentificationPair: manager.ProbeIdentificationPair{
+				EBPFFuncName: connWriteRetProbe,
+			},
+		},
+		{
+			ProbeIdentificationPair: manager.ProbeIdentificationPair{
+				EBPFFuncName: connCloseProbe,
+			},
+		},
+	},
+}
 
-func newGoTLSProgram(c *config.Config, sockFD *ebpf.Map) *GoTLSProgram {
-	if !c.EnableGoTLSSupport {
-		return nil
-	}
+func newGoTLSProgramProtocolFactory(m *manager.Manager, sockFDMap *ebpf.Map) protocols.ProtocolFactory {
+	return func(c *config.Config) (protocols.Protocol, error) {
+		if !c.EnableGoTLSSupport {
+			return nil, nil
+		}
 
-	if !http.HTTPSSupported(c) {
-		log.Errorf("goTLS not supported by this platform")
-		return nil
-	}
+		if !http.HTTPSSupported(c) {
+			log.Errorf("goTLS not supported by this platform")
+			return nil, nil
+		}
 
-	if !c.EnableRuntimeCompiler && !c.EnableCORE {
-		log.Errorf("goTLS support requires runtime-compilation or CO-RE to be enabled")
-		return nil
-	}
+		if !c.EnableRuntimeCompiler && !c.EnableCORE {
+			log.Errorf("goTLS support requires runtime-compilation or CO-RE to be enabled")
+			return nil, nil
+		}
 
-	return &GoTLSProgram{
-		done:              make(chan struct{}),
-		cfg:               c,
-		procRoot:          c.ProcRoot,
-		sockFDMap:         sockFD,
-		binAnalysisMetric: libtelemetry.NewCounter("usm.go_tls.analysis_time", libtelemetry.OptPrometheus),
-		registry:          utils.NewFileRegistry("go-tls"),
+		return &goTLSProgram{
+			done:              make(chan struct{}),
+			cfg:               c,
+			manager:           m,
+			procRoot:          c.ProcRoot,
+			sockFDMap:         sockFDMap,
+			binAnalysisMetric: libtelemetry.NewCounter("usm.go_tls.analysis_time", libtelemetry.OptPrometheus),
+			registry:          utils.NewFileRegistry("go-tls"),
+		}, nil
 	}
 }
 
 // Name return the program's name.
-func (p *GoTLSProgram) Name() string {
+func (p *goTLSProgram) Name() string {
 	return "go-tls"
 }
 
 // IsBuildModeSupported return true if the build mode is supported.
-func (*GoTLSProgram) IsBuildModeSupported(mode buildmode.Type) bool {
+func (*goTLSProgram) IsBuildModeSupported(mode buildmode.Type) bool {
 	return mode == buildmode.CORE || mode == buildmode.RuntimeCompiled
 }
 
-// ConfigureManager adds maps to the given manager.
-func (p *GoTLSProgram) ConfigureManager(m *errtelemetry.Manager) {
-	p.manager = m
-	p.manager.Maps = append(p.manager.Maps, []*manager.Map{
-		{Name: offsetsDataMap},
-		{Name: goTLSReadArgsMap},
-		{Name: goTLSWriteArgsMap},
-		{Name: connectionTupleByGoTLSMap},
-	}...)
-	// Hooks will be added in runtime for each binary
-}
-
 // ConfigureOptions changes map attributes to the given options.
-func (p *GoTLSProgram) ConfigureOptions(options *manager.Options) {
+func (p *goTLSProgram) ConfigureOptions(m *manager.Manager, options *manager.Options) {
```

Comment:
`m` is not used, it will trigger a linter error
```suggestion
func (p *goTLSProgram) ConfigureOptions(_ *manager.Manager, options *manager.Options) {
```